From: mw at semihalf.com (Marcin Wojtas)
Date: Mon, 10 May 2021 17:11:15 +0200
Subject: [PATCH] acpi: spcr: accept baud rate 0x0 as a preconfigured state
Message-ID: <20210510151115.792916-1-mw@semihalf.com>

Since SPCR 1.04 [1] the baud rate of 0x0 means a preconfigured state of UART.
Assume the firmware or bootloader configures console correctly and accept
this value as a valid one in the SPCR ACPI test.

[1]: https://docs.microsoft.com/en-us/windows-hardware/drivers/serports/serial-port-console-redirection-table

Signed-off-by: Marcin Wojtas <mw at semihalf.com>
---
 src/acpi/spcr/spcr.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/acpi/spcr/spcr.c b/src/acpi/spcr/spcr.c
index 2dbb9533..dda309f2 100644
--- a/src/acpi/spcr/spcr.c
+++ b/src/acpi/spcr/spcr.c
@@ -177,6 +177,9 @@ static int spcr_test1(fwts_framework *fw)
 
 	reserved = false;
 	switch (spcr->baud_rate) {
+	case 0x00:
+		str = "preconfigured in firmware";
+		break;
 	case 0x03:
 		str = "9600";
 		break;
-- 
2.29.0

